src/geo/cartodb-layer-group.js

```diff
@@ -131,10 +139,18 @@ var CartoDBLayerGroup = Backbone.Model.extend({
 
   onLayerVisibilityChanged: function (callback) {
     this._layersCollection.on('change:visible', function (layerModel) {
-      if (this.getLayers().indexOf(layerModel) >= 0) {
+      if (this.contains(layerModel)) {
         callback(layerModel);
```

Comment:
Off-topic: one thing I usually do is always `return callback(...);`, so if later I add other branches to the code it will not call callback more than once.